platform/features/table/src/Table.js

```diff
@@ -0,0 +1,163 @@
+/*****************************************************************************
+ * Open MCT Web, Copyright (c) 2014-2015, United States Government
+ * as represented by the Administrator of the National Aeronautics and Space
+ * Administration. All rights reserved.
+ *
+ * Open MCT Web is licensed under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * http://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ * Open MCT Web includes source code licensed under additional open source
+ * licenses. See the Open Source Licenses file (LICENSES.md) included with
+ * this source code distribution or the Licensing information page available
+ * at runtime from the About dialog for additional information.
+ *****************************************************************************/
+/*global define,moment*/
+
+define(
+    [
+        './DomainColumn',
+        './RangeColumn',
+        './NameColumn'
+    ],
+    function (DomainColumn, RangeColumn, NameColumn) {
+        "use strict";
+
+        /**
+         * Class that manages table metadata, state, and contents.
+         * @memberof platform/features/table
+         * @param domainObject
+         * @constructor
+         */
+        function Table(domainObject, telemetryFormatter) {
```

Comment:
re: TableConfiguration. I'm a little concerned that it overloads the term 'configuration'. But I agree 'Table' doesn't give you much of a clue about what it does.

I debated where responsibility for object model configuration should reside. To be honest I'm not sure the Table class should be concerned with object model configuration at all. Having a concept of 'column configuration' is fine, but it might be better if it's agnostic of the source of that configuration, and just have it handed to it. That said, it simplifies the controller if the Table class handles both reading and writing model configuration, and simplification was the original motivation for creating the Table class at all.